exp/cloud/converters/vmss.go

```diff
@@ -0,0 +1,51 @@
+package converters
+
+import (
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/compute/mgmt/compute"
+	"github.com/Azure/go-autorest/autorest/to"
+
+	capzv1 "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha3"
+	capzv1convert "sigs.k8s.io/cluster-api-provider-azure/cloud/converters"
+	capzv1exp "sigs.k8s.io/cluster-api-provider-azure/exp/api/v1alpha3"
+)
+
+// SDKToVMSS converts an Azure SDK VirtualMachineScaleSet to the AzureMachinePool type.
+func SDKToVMSS(sdkvmss compute.VirtualMachineScaleSet, sdkinstances []compute.VirtualMachineScaleSetVM) (*capzv1exp.VMSS, error) {
```

Comment:
I encountered a similar issue with AKS and wasn't sure if I wanted to add a new type or pass around the Azure types. Do you have any opinions on why introducing a new type is better? I'm inclined to agree, but I dislike that we then need three types (VMSS, Azure VMSS, and the Spec), to describe the same object in slightly different ways.

Reply:
💯 -- please take my upvotes. I have but one.

I think the middle type is not needed, but I would like to take that up in a design / refactoring proposal / PR. Would be great to work together on it.